### DISCUSSION TITLE
Is there a way to set the vertical carousel height based on the slide height?

### DISCUSSION THREAD

**Discussion by NickWoodward** [davidjerleke/embla-carousel#566] [Help] [answered]
So for example I'd like to be able to not define a container height and have it wrap to the slide height? Thanks!

** edit: If that's not possible, can I achieve something like this, but without the spacing between the slides? I'm struggling with how the slides are styled within the container, it seems like the container doesn't actually behave like a wrapper?

https://codesandbox.io/s/embla-carousel-arrow-dots-react-forked-lszlrz?file=/src/css/embla.css:0-2686

**Comment by davidjerleke** [comment, marked as answer]
Hi @NickWoodward,

No you have to specify a height for a vertical carousel. That’s how CSS work - It’s a browser limitation for containers with vertical overflow content. Please [read this](https://www.embla-carousel.com/guides/slide-container/#using-flexbox).

However, why don’t you use the [carousel generator](https://www.embla-carousel.com/examples/generator/) and generate a vertical carousel? That will show you how to setup your CSS in a correct way.

Best,
David

**Reply by davidjerleke** [reply]
@NickWoodward did you see [my response](https://github.com/davidjerleke/embla-carousel/discussions/566#discussioncomment-6953621) and try the [carousel generator](https://www.embla-carousel.com/examples/generator/)?

**Reply by NickWoodward** [reply]
> @NickWoodward did you see [my response](https://github.com/davidjerleke/embla-carousel/discussions/566#discussioncomment-6953621) and try the [carousel generator](https://www.embla-carousel.com/examples/generator/)?

oh sorry @davidjerleke, thought I'd already replied. my mistake, but yes it was very helpful, thank you
